### PR TITLE
#512: Make DeploymentConfig a scalable resource.

### PR DESCRIPTION
**Pull request by iocanel**

### CODE REVIEW

File: openshift-client/src/main/java/io/fabric8/openshift/client/dsl/internal/DeploymentConfigOperationsImpl.java [hunk 1]
```diff
@@ -115,4 +116,19 @@ public void run() {
...
+  @Override
+  public DeploymentConfig scale(int count) {
+    return scale(count, false);
+  }
+
+  @Override
+  public DeploymentConfig scale(int count, boolean wait) {
+    DeploymentConfig res = cascading(false).edit().editSpec().withReplicas(count).endSpec().done();
+    if (wait) {
+      res = getMandatory();
```
Comment (jimmidyson) [inline openshift-client/src/main/java/io/fabric8/openshift/client/dsl/internal/DeploymentConfigOperationsImpl.java]: Shouldn't this wait until scaled? Is this what we do elsewhere?

Comment (jimmidyson) [inline openshift-client/src/main/java/io/fabric8/openshift/client/dsl/internal/DeploymentConfigOperationsImpl.java]: e.g. https://github.com/fabric8io/kubernetes-client/blob/35dda57692936d9eb429aa909c2d29f6877cd547/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/DeploymentOperationsImpl.java#L65

Comment (iocanel) [inline openshift-client/src/main/java/io/fabric8/openshift/client/dsl/internal/DeploymentConfigOperationsImpl.java]: Sure, but as mentioned on #512 we can't wait for DeploymentConfig, as we don't have access to the replicas from the status.

**Comment by christian-posta** [comment]
would love to have this today if possible for a talk I'm doing tonight :)

**Comment by jstrachan** [comment]
not having this is a major usability issue for openshift with fabric8-maven-plugin (start/stop/scale doesn't work ;)

File: openshift-client/src/main/java/io/fabric8/openshift/client/dsl/internal/DeploymentConfigOperationsImpl.java (hunk 1 above)
Comment (jimmidyson) [inline openshift-client/src/main/java/io/fabric8/openshift/client/dsl/internal/DeploymentConfigOperationsImpl.java]: Check out the latest model - should be available now iirc.

Comment (iocanel) [inline openshift-client/src/main/java/io/fabric8/openshift/client/dsl/internal/DeploymentConfigOperationsImpl.java]: Seems that it is. Awesome!

**Comment by jstrachan** [comment]
@jimmidyson you cool with the latest changes from @iocanel ?

**Comment by jimmidyson** [comment]
LGTM [merge]

**Pull request merged**